core/src/main/java/org/apache/iceberg/util/ArrayUtil.java

```diff
@@ -300,4 +310,14 @@ private static Object copyArrayGrow1(final Object array, final Class<?> newArray
     }
     return Array.newInstance(newArrayComponentType, 1);
   }
+
+  public static boolean isStrictlyOrdered(long[] array) {
+    for (int index = 1; index < array.length; index++) {
```

Comment:
This method only returns true when the array is strictly ascending ordered.  It will return false when the array is strictly descending ordered.
Maybe `isStrictlyAscendingOrdered` is a better method. WDYT?

Reply:
That's a good point. I renamed it to `isStrictlyAscending`.